distributed/spill.py

```diff
@@ -240,8 +344,47 @@ def spilled_total(self) -> SpilledSize:
         The two may differ substantially, e.g. if sizeof() is inaccurate or in case of
         compression.
         """
-        slow = cast(zict.Cache, self.slow).data if has_zict_220 else self.slow
-        return cast(Slow, slow).total_weight
+        return self._slow_uncached.total_weight
+
+    def get_metrics(self) -> dict[str, float]:
+        """Metrics to be exported to Prometheus or to be parsed directly.
+
+        From these you may generate derived metrics:
+
+        cache hit ratio:
+          by keys  = memory_read_count_total / (memory_read_count_total + disk_read_count_total)
+          by bytes = memory_read_bytes_total / (memory_read_bytes_total + disk_read_bytes_total)
+
+        mean times per key:
+          pickle   = pickle_time_total     / disk_write_count_total
+          write    = disk_write_time_total / disk_write_count_total
+          unpickle = unpickle_time_total   / disk_read_count_total
+          read     = disk_read_time_total  / disk_read_count_total
+
+        mean bytes per key:
+          write    = disk_write_bytes_total / disk_write_count_total
+          read     = disk_read_bytes_total  / disk_read_count_total
+
+        mean bytes per second:
+          write    = disk_write_bytes_total / disk_write_time_total
+          read     = disk_read_bytes_total  / disk_read_time_total
+        """
+        fm = self.fast_metrics
+        sm = self._slow_uncached.metrics
+
+        out = {
+            "count_max": self.count_max,
+            "bytes_max": self.bytes_max,
+            "memory_count": len(self.fast),
+            "memory_bytes": self.fast.total_weight,
+            "disk_count": len(self.slow),
+            "disk_bytes": self._slow_uncached.total_weight.disk,
+        }
+        for k, v in fm.__dict__.items():
+            out[f"memory_{k}"] = v
+        for k, v in sm.__dict__.items():
+            out[k if "pickle" in k else f"disk_{k}"] = v
+        return out
```

Comment:
Historically, this kind of measure was both exported to Prometheus and sent to the scheduler through the heartbeat.
This is a dict with 26 keys, which makes it somewhat substantial. I don't think that blindly adding more and more data to the heartbeat is a good design - particularly as we're not using it in the Bokeh dashboard.

@fjetter opinion?

Reply:
What are the 26 keys?

Reply:
Yes, don't submit over the heartbeat unless it is actually used there, please.